### ISSUE TITLE
Idea/Future: More declarative setup by `install`

### ISSUE THREAD

**Issue by andreas-kupries** [epinio/epinio#45]
From the chat ...

@jimmykarily I keep thinking that what we are doing right now could be done in a better way. Right now we are just translating bash commands to go commands in the same "procedural" way.
After we are done we should consider moving to a more "descriptive" way of installing things. E.g. each deployment declares the "conditions" that should be met before the installation starts. We could then fire all of them in parallel and let them do their magic.
This would enter the realm of other projects though (e.g. what Matt presented yesterday) so I'm not sure we want to go there.

@andreas-kupries That might then also allow for continuing a partial install, i.e. skip the the pieces already installed, and run only the missing parts.

**Comment by jimmykarily** [comment]
Since writing this issue, we've switched to a parallel installation. Also, each component waits for it's dependencies to appear (e.g. deployments that need a cert, will wait for cert-manager to be there). In a way, we've implemented some of the ideas here although, it's not completely declarative and we didn't achieve idempotent installation.

This issue is mixing different ideas (parallel install, declarative installation, idempotent installation) and some parts are already implemented. We should open new issues for the non-implemented features if we need them. Closing.
